addOns/savehar/src/main/java/org/zaproxy/zap/extension/savehar/PopupMenuItemSaveHarMessage.java

```diff
@@ -20,28 +20,45 @@
 package org.zaproxy.zap.extension.savehar;
 
 import edu.umass.cs.benchlab.har.HarEntries;
+import edu.umass.cs.benchlab.har.HarEntry;
 import edu.umass.cs.benchlab.har.HarLog;
+import java.awt.Component;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
 import java.io.BufferedOutputStream;
 import java.io.File;
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.io.OutputStream;
 import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.List;
 import javax.swing.JFileChooser;
+import javax.swing.JTree;
 import javax.swing.filechooser.FileFilter;
+import javax.swing.tree.TreePath;
+import org.apache.commons.lang.ArrayUtils;
 import org.apache.log4j.Logger;
 import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.db.DatabaseException;
+import org.parosproxy.paros.extension.Extension;
+import org.parosproxy.paros.extension.ExtensionPopupMenuItem;
+import org.parosproxy.paros.model.HistoryReference;
 import org.parosproxy.paros.model.Model;
-import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.model.SiteNode;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
 import org.parosproxy.paros.view.View;
 import org.zaproxy.zap.utils.HarUtils;
-import org.zaproxy.zap.view.popup.PopupMenuItemHttpMessageContainer;
 import org.zaproxy.zap.view.widgets.WritableFileChooser;
 
-public class PopupMenuItemSaveHarMessage extends PopupMenuItemHttpMessageContainer {
+public class PopupMenuItemSaveHarMessage extends ExtensionPopupMenuItem {
```

Comment:
It would be better to keep using `PopupMenuItemHttpMessageContainer` as that applies to more UI components (any that has messages). It needs `multiSelect` set to true when calling the base constructor though.

One can override `performActions(List<HttpMessage>)` to save all of them at the same time. (It's possible that not all of them have been persisted, in that case we just not include the custom fields.)